### ISSUE TITLE
Is is possible to set a base path (relative or absolute) for rendering images inserted into the editor?

### ISSUE THREAD

**Issue by fevangelou** [Alex-D/Trumbowyg#1075]
I'm building a CMS plugin that utilizes Trumbowyg.

The CMS has a discrete backend, much like WordPress does (/wp-admin/).

When inserting images from the CMS into the editor, the supplied path of these images **by the CMS** is relative to the site's frontend.

That means images are inserted with paths like "assets/images/X.jpg" when the backend admin is located in "/admin/app/whatever/".

The browser then attempts to load the image from the combined path "/admin/app/whatever/assets/images/X.jpg" which does not exist obviously, again, because the CMS inserts media using relative paths to the frontend.

See the rendered HTML (from Trumbowyg) below:
![Screen Shot 2020-02-03 at 20 58 07](https://user-images.githubusercontent.com/1301787/73681947-0173ef00-46c8-11ea-93bb-ac2d85d544e3.png)

Is there some option in Trumbowyg to enforce a base path for any media inserted into the editor, so that they are properly rendered from their actual location, regardless of where Trumbowyg is loaded on?

Thank you in advance.

**Comment by Alex-D** [comment]
There is no option for that in Trumbowyg.

You can use the `<base>` HTML tag: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/base

**Comment by fevangelou** [comment]
Adjusting the base tag would break the CMS :)

The point is to only set a base path for the rendered content inside the editor...

**Comment by Alex-D** [comment]
The right way to do that is to send the right path from the back-end, not to tweak the front ^^

**Comment by fevangelou** [comment]
Actually, it's possible.

Instead of hooking up Trumbowyg directly to the CMS's textarea field, I created an intermediate div element with the contenteditable attribute. On form load, the data is copied from the textarea (which is hidden from display) to the contenteditable div, and after that, whatever is written/changed on the div is sync'ed back to the textarea.

Then I hooked up Trumbowyg to the contenteditable div and since I controlled the data entry (listener on the "input" event), I was able to dynamically replace the media file paths (for images, iframes, video/audio tags) with absolute ones from the correct location, just for preview purposes. The data sync'ed back to the textarea maintain relative paths (so the content is also compatible with other editor plugins of that CMS).

This also allowed me to be able to properly insert content from other widgets of the CMS that are provided for the editor plugin (e.g. options to insert direct links to content articles, image galleries and so on). So far I haven't seen any conflicts of this implementation with Trumbowyg.

I'm just leaving this here as an idea in case anyone else stumbles on the same issue as I did.

Alexandre, thank you for your time and effort building Trumbowyg.

Merci :)
